src/lib/HsToCoq/Util/GHC/Module.hs

```diff
@@ -1,16 +1,28 @@
 {-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# LANGUAGE TemplateHaskell #-}
 
-module HsToCoq.Util.GHC.Module (module Module, moduleNameText, mkModuleNameT) where
+module HsToCoq.Util.GHC.Module (module Module, ModuleData(..), moduleNameText, mkModuleNameT, modName, modExports) where
 
 import Module
 
 import Data.Text (Text)
 import qualified Data.Text as T
 
+import qualified GHC
+
+import Control.Lens
+
+data ModuleData = ModuleData { _modName     :: ModuleName
+                             , _modExports  :: [GHC.Name]
```

Comment:
I suppose `_modExports` represents all the functions/constructors exported in this module. In that case, are these names before or after renaming? Do they contain skipped functions/constructors? Is there a way to tell if a name is a function or a constructor?

Reply:
Yes, `_modExports` represents all names exported by the module, which could include functions and constructors, as well as datatypes and type classes. The names are taken from the Haskell source files using the GHC API, so I don't think any edits apply. In other words, the names are before renaming, and skipped functions and constructors are included. Could this be problematic?

Currently, there isn't a way to tell whether a name is a function or a constructor, but I could work on this if it would be helpful.

Reply:
Great! The main reason I was asking was for clarification. I think it may be helpful to document the fact that these names are before renaming as a comment. It would be helpful to other people who are going to working on it, and it would also be helpful to us as one day we might forget that :)

I don't know if it's worth it to implement a function to tell whether a name is a function or a constructor. If you don't need that function at this point, it's totally fine to leave it as it is now. Or you can do that if it's easy.

Reply:
> I think it may be helpful to document the fact that these names are before renaming as a comment.

Sure, I will add a comment for this.

> I don't know if it's worth it to implement a function to tell whether a name is a function or a constructor. If you don't need that function at this point, it's totally fine to leave it as it is now. Or you can do that if it's easy.

At this point, I don't think I need that ability, but we could add it in the future if the need arises.

Reply:
Great. Can you push a version with the comments? I will merge the PR once that is done.

Reply:
Done!

Reply:
Thanks!